circup/command_utils.py

```diff
@@ -841,3 +933,34 @@ def sorted_by_directory_then_alpha(list_of_files):
         sorted_full_list.append(files[cur_name])
 
     return sorted_full_list
+
+
+def find_pyproject():
+    """
+    Look for a pyproject.toml in the current directory or its parent directories.
+
+    :return: The path to the pyproject.toml for the project, or None if it
+        couldn't be found.
+    """
+    logger.info("Looking for pyproject.toml file.")
+    cwd = Path.cwd()
+    candidates = [cwd]
+    candidates.extend(cwd.parents)
```

Comment:
This extends the search all the way back to the root of the system (on linux), I think that is a bit of an extreme length to go looking for a pyproject.toml file. 

I would be in favor of limiting this to either the direct parent only, or maybe 1 level above that if you feel its warranted.

If very deeply nested project structures beyond that are a concern perhaps we could also just allow the pyproject.toml file path to be specified by the user with a command line argument so that we aren't searching around too many places for it.